src/type_checkers/use.cr

```diff
@@ -22,7 +22,7 @@ module Mint
 
       # This is checked by the provider so we assume it's there
       subscription =
-        records.find!(&.name.==(provider.subscription))
+        records.find!(&.name.==(provider.subscription.value))
```

Comment:
`#name` is not a `TypeId`?

```suggestion
        records.find!(&.name.value.==(provider.subscription.value))
```

Reply:
Actually not in this case :smile:!  `records` here is a list of `TypeChecker::Record`'s, where `name` is a [`String`](https://github.com/mint-lang/mint/blob/master/src/type_checker/record.cr#L8)